Add support for Sentry native crashes and allow users to disable it or use their own DSN with client settings.

Pombal opened the pull request:
Upload debug symbols for nightly builds.
Add documentation for Ambient settings.

FredrikNoren reviewed (approved):
Nice!
For the settings doc to appear you need to update the SUMMARY.md file as well. 
Also there's a .github/release workflow which will also need the debug symbols (which we use to release point versions of ambient)

Pombal commented:
@FredrikNoren Good catch!
I've updated the docs and moved the symbol upload to .github/release. I thought we were going to do releases from nightly builds and missed release.yml.

philpax commented on crates/settings/src/general.rs:
```diff
@@ -0,0 +1,30 @@
+
+// enabled, dsn
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Sentry(pub (bool, String));
```
I'd use explicit fields for this to make it obvious what they represent, especially for people editing the config

philpax commented on docs/src/user/settings.md:
```diff
@@ -0,0 +1,14 @@
+
+## Settings
+
+[general]\
```
You can use three-backticks followed by toml here, like here https://github.com/AmbientRun/Ambient/blob/main/docs/src/reference/models.md?plain=1#L8-L12

philpax reviewed (commented):
LGTM, but a few minor tweaks before merge
